MoneySaver.Api.Services/Implementation/TransactionCategoryService.cs

```diff
@@ -23,10 +23,18 @@ public TransactionCategoryService(IRepository<TransactionCategory> categoryRepos
             this.mapper = mapper;
         }
 
-        public TransactionCategoryModel CreateCategory(TransactionCategoryModel categoryModel)
+        public async Task<TransactionCategoryModel> CreateCategoryAsync(TransactionCategoryModel categoryModel)
         {
-            TransactionCategory transactionCategory = mapper.Map<TransactionCategory>(categoryModel);
-            this.categoryRepository.AddAsync(transactionCategory);
+            try
+            {
+                TransactionCategory transactionCategory = mapper.Map<TransactionCategory>(categoryModel);
+                await this.categoryRepository.AddAsync(transactionCategory);
```

Comment:
The id value from the newly created CategoryTransaction should be set in the TransactionCategoryModel too, before return.

Reply:
The Mapper should set the TransactionCategoryId out of the box, or I am wrong? Feel free to correct me.